### PR TITLE
Signatures

### PR DESCRIPTION
**Pull request by zwade**
Adds support for getting the type of signatures as well as drilling in on the type of expressions

Shall we try again?

### CODE REVIEW

File: merlin/process.py [hunk 1]
```diff
@@ -108,10 +107,12 @@ def send_command(self, cmd):
         if self.mainpipe is None or self.mainpipe.returncode is not None:
             self.restart()
 
+        print(cmd)
```
Comment (bluepichu) [inline merlin/process.py]: Pls no thx

File: merlin/process.py [hunk 2]
```diff
@@ -108,10 +107,12 @@ def send_command(self, cmd):
...
         self.mainpipe.stdin.write(json.dumps(cmd).encode('utf-8'))
         self.mainpipe.stdin.flush()
         line = self.mainpipe.stdout.readline()
         result = json.loads(line.decode('utf-8'))
+        print(result)
```
Comment (bluepichu) [inline merlin/process.py]: Pls no again thx

File: merlin/process.py [hunk 3]
```diff
@@ -143,13 +144,13 @@ def __init__(self, process, view):
         self.process = process
         self.view = view
 
-    def send_query(self, *query):
+    def send_query(self, *query, verbosity=None):
```
Comment (bluepichu) [inline merlin/process.py:144]: If verbosity isn't used in other commands, any reason it couldn't just be part of the query?

File: sublime-text-merlin.py [hunk 4]
```diff
@@ -225,29 +239,68 @@ def _item_format(self, item):
...
+
+    def update_enclosing(self):
+        pos = self.view.sel()
+        line, col = self.view.rowcol(pos[0].begin())
+        print(self.verbosity)
+        enclosing = self.merlin.type_enclosing(line + 1, col, verbosity=self.verbosity)
+        self.enclosing = enclosing 
+
+    def show_deepen(self):
+        print("Deepen")
```
Comment (bluepichu) [inline sublime-text-merlin.py]: Pls no again again thx

File: sublime-text-merlin.py [hunk 5]
```diff
@@ -261,27 +314,57 @@ class MerlinTypeAtCursorCmd(sublime_plugin.WindowCommand):
...
 
 
 class MerlinTypeMenu(sublime_plugin.TextCommand):
     """
     Display type information in context menu
     """
     def run(self, edit):
-        enclosing = MerlinTypeEnclosing(self.view)
-        enclosing.show_menu()
+        enclosing[self.view.id()] = MerlinTypeEnclosing(self.view)
```
Comment (bluepichu) [inline sublime-text-merlin.py:371]: Not critical for this PR: maybe enclosing should be a field of the merlin view?

**Review by bluepichu — CHANGES_REQUESTED** [review]
Pls no print thx

File: merlin/process.py (hunk 3 above)
Comment (zwade) [reply merlin/process.py:144]: So verbosity is used in the other commands, I just haven't figured out how. Also verbosity isn't part of the query object, its external to that

File: sublime-text-merlin.py (hunk 5 above)
Comment (zwade) [reply sublime-text-merlin.py:371]: perhaps, i'll file that as todo

**Review by bluepichu — APPROVED** [review]

**Pull request merged**